var/spack/repos/builtin/packages/sgpp/package.py

```diff
@@ -72,11 +72,13 @@ class Sgpp(SConsPackage):
     extends("python", when="+python")
     depends_on("py-pip", when="+python", type="build")
     depends_on("py-wheel", when="+python", type="build")
-    # TODO allow newer versions once distutils is removed from SGpp
-    depends_on("py-setuptools@:59", type=("build"))
-    # TODO allow newer versions once distutils is removed from SGpp
-    depends_on("python@3.7:3.11", type=("build", "run"))
-    depends_on("swig@3:", when="+python", type=("build"))
+    depends_on("py-setuptools", type=("build"))
+    # Older SGpp releases (:3.4.0) do not support python 3.12 due to them using distutils
+    depends_on("python@3.7:3.11", type=("build", "run"), when="@:3.4.0")
+    # SGpp@master works with newer python versions (3.12:) as well
+    depends_on("python@3.7:", type=("build", "run"))
+    # Newest swig version 4.1 seems to cause problem -> limit to 3:4.0 for now
+    depends_on("swig@3:4.0", when="+python", type=("build"))
```

Comment:
```suggestion
    depends_on("swig@3:4.0", when="+python", type="build")
```

Reply:
Added in [95f303a](https://github.com/spack/spack/pull/43178/commits/95f303ac719db4fd8126748bd06414ed40d15551)